modern/src/reports/ChartReportPage.js

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+import { Box, Paper } from '@material-ui/core';
+import ReportFilter from './ReportFilter';
+import ReportLayoutPage from './ReportLayoutPage';
+import { useAttributePreference } from '../common/preferences';
+import { getConverter, formatDate } from '../common/formatter';
+import {LineChart, Line, XAxis, YAxis, CartesianGrid, Tooltip, Legend, ResponsiveContainer } from 'recharts';
+
+const ReportFilterForm = ({ setItems }) => {
+
+  const speedUnit = useAttributePreference('speedUnit');
+
+  const handleSubmit = async (deviceId, from, to, mail, headers) => {
+    const query = new URLSearchParams({ deviceId, from, to, mail });
+    const response = await fetch(`/api/reports/route?${query.toString()}`, { headers });
+    if (response.ok) {
+      const data = await response.json();
+      let formattedData = data.map((obj)=>{
+        return Object.assign(obj, 
+                              {speed: Number(getConverter('speed')(obj.speed, speedUnit))},
+                              {fixTime: formatDate(obj.fixTime)}
+                            );
+      })
+      setItems(formattedData);
+    }
+  }
+
+  return (
+    <ReportFilter handleSubmit={handleSubmit} showOnly />
+  )
+};
+
+const CustomizedAxisTick = ({ x, y, payload }) =>{
+  const parts = payload.value.split(' ');
+  return (
+    <g transform={`translate(${x},${y})`}>
+      <text x={0} y={0} dy={16} textAnchor="end" fill="#666" transform="rotate(-35)">{parts[0]}</text>
+      <text x={0} y={16} dy={16} textAnchor="end" fill="#666" transform="rotate(-35)">{parts[1]}</text>
+    </g>
+  );
+}
+
+const ChartReportPage = () => {
+
+  const [items, setItems] = useState([]);
+  const [type, setType] = useState('speed');
+
+  return (
+    <ReportLayoutPage reportFilterForm={ReportFilterForm} setItems={setItems} setType={setType} showChartType>
+      <Paper>
+        <Box height={400}>
+          <ResponsiveContainer>
+            <LineChart data={items}>
+              <XAxis dataKey="fixTime" tick={<CustomizedAxisTick/>} height={60} />
```

Comment:
There are some hardcoded numbers like height and grid. Is there a way to get rid of them? Can you please clarify what each one is for.

Reply:
Box->height is for setting chart height. I have tried setting the chart height using CSS styling but it's not working.

rest of the hardcoded style is chart specific. Xaxis->height is for setting x-axis height so that date and time can be rotated and still visible.